src/Libraries/CoreNodesUI/Input/DoubleSlider.cs

```diff
@@ -1,113 +1,95 @@
         {
-            get { return _max; }
-            set
-            {
-                _max = value;
-                if (_max < Value)
-                    Value = _max;
-                RaisePropertyChanged("Max");
-            }
+            var rhs = AstFactory.BuildDoubleNode(Value);
+            var assignment = AstFactory.BuildAssignment(GetAstIdentifierForOutputIndex(0), rhs);
+
+            return new[] { assignment };
         }
 
-        private double _min;
-        public double Min
+        protected override double DeserializeValue(string val)
         {
-            get { return _min; }
-            set
+            try
             {
-                _min = value;
-                if (_min > Value)
-                    Value = _min;
-                RaisePropertyChanged("Min");
+                return Convert.ToDouble(val, CultureInfo.InvariantCulture);
             }
-        }
-
-        #region Serialization/Deserialization Methods
-
-        protected override void SerializeCore(XmlElement element, SaveContext context)
-        {
-            base.SerializeCore(element, context); // Base implementation must be called.
-
-            XmlElement outEl = element.OwnerDocument.CreateElement("Range");
-            outEl.SetAttribute("min", Min.ToString(CultureInfo.InvariantCulture));
-            outEl.SetAttribute("max", Max.ToString(CultureInfo.InvariantCulture));
-            element.AppendChild(outEl);
-        }
-
-        protected override void DeserializeCore(XmlElement nodeElement, SaveContext context)
-        {
-            base.DeserializeCore(nodeElement, context); //Base implementation must be called.
-
-            foreach (XmlNode subNode in nodeElement.ChildNodes)
+            catch
             {
-                if (!subNode.Name.Equals("Range"))
-                    continue;
-
-                double min = Min;
-                double max = Max;
-
-                if (subNode.Attributes != null)
-                {
-                    foreach (XmlAttribute attr in subNode.Attributes)
-                    {
-                        if (attr.Name.Equals("min"))
-                            min = Convert.ToDouble(attr.Value, CultureInfo.InvariantCulture);
-                        else if (attr.Name.Equals("max"))
-                            max = Convert.ToDouble(attr.Value, CultureInfo.InvariantCulture);
-                        else if (attr.Name.Equals("value"))
-                            Value = Convert.ToDouble(subNode.InnerText, CultureInfo.InvariantCulture);
-                    }
-                }
-
-                Min = min;
-                Max = max;
+                return 0;
             }
         }
 
-        #endregion
+        protected override string SerializeValue()
+        {
+            return Value.ToString(CultureInfo.InvariantCulture);
+        }
 
         protected override bool UpdateValueCore(string name, string value, UndoRedoRecorder recorder)
```

Comment:
Nearly all the code in `UpdateValueCore` method of `DoubleSlider` and `IntegerSlider` are the same. Since `SliderBase` is a templated base class with a specific type, can't we move these to `SliderBase`? Perhaps if something were to be different, that part can be made `protected virtual` and implemented here.